webapp/lib/controller.dart

```diff
@@ -204,21 +212,37 @@ bool isProjectStale(model.NeedsReplyData projectData) {
   }
 }
 
-void setupProjectTimer(model.NeedsReplyData projectData, [bool stale = false]) {
-  projectTimers[projectData.project]?.cancel();
+String getProjectTimer(Object data) {
+  if (data is model.NeedsReplyData) {
+    return  (data as model.NeedsReplyData).project;
+  } else if(data is model.SystemMetricsData) {
+    return SYSTEM_METRICS_ROOT_COLLECTION_KEY;
+  }
```

Comment:
consider throwing an error if data is neither of the two options? just for sanity checking